cgp/local_search/evolution_strategies.py

```diff
@@ -0,0 +1,218 @@
+import functools
+import math
+import multiprocessing as mp
+from typing import TYPE_CHECKING, Callable, Dict, List, Tuple, Union
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from ..individual import IndividualBase
+
+
+class EvolutionStrategies:
+    def __init__(
+        self,
+        objective: Callable[[Callable[["np.ndarray[float]"], "np.ndarray[float]"]], float],
+        seed: int,
+        *,
+        learning_rate_mu: float = 1.0,
+        learning_rate_sigma: Union[None, float] = None,
+        population_size: Union[None, int] = None,
+        max_generations: int = 10,
+        min_sigma: float = 1e-6,
+        fitness_shaping: bool = True,
+        mirrored_sampling: bool = True,
+        n_processes: int = 1
+    ) -> None:
+        """Evolution strategies using the natural gradient of multinormal
+        search distributions in natural coordinates.
+
+        Standard deviation of the search distribution for each
+        parameter is stored per individual. Offspring may use the
+        values of their parents if the number of used parameters has
+        not changed. Does not consider covariances between parameters.
+
+        Implementation and default values following Wierstra et
+        al. (2014). Natural evolution strategies. Journal of Machine
+        Learning Research, 15(1), 949-980.
+
+        Parameters
+        ----------
+        individual : Individual
+            Individual for which to perform local search.
+        objective : Callable[[Callable[[np.ndarray[float]], np.ndarray[float]]], float]
+            Objective function used for the local search. Needs to accept
+            an numpy-compatible function as first argument (e.g., from
+            IndividualBase.to_numpy()) and return a float representing the
+            fitness of the individual.
+        seed : int
+            Seed for internal random number generator.
+        learning_rate_mu : float, optional
+            Learning rate for mean of search distribution.
+            Defaults to 1.0.
+        learning_rate_sigma : float, optional
+            Learning rate for standard deviation of search
+            distribution. Defaults to (3 + log(dim)) / (5.0 *
+            sqrt(dim)), where dim is the dimension of the search space.
```

Comment:
The default in the docstring is different to the one above?

Reply:
yes, the default depends on the dimensionality of the search space, so we initially set it to `None`, then check for `None` below and compute the default if necessary (e.g., l115ff)